From a452173ed5a68d31c9a7a7dc10212eabbd9e073a Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 6 Nov 2018 11:36:52 +0100
Subject: [PATCH] build: Fix build order with libtracker-sparql generated
 headers

Both libtracker-fts and libtracker-data end up requiring includes
from tracker-sparql.h, but we have to generate tracker-generated.h
first.

It's pretty terrible to add intermediate targets as a dependency
outside the libtracker-sparql directory, but it doesn't seem we
can do better.

Closes: https://gitlab.gnome.org/GNOME/tracker/issues/52
---
 src/libtracker-data/meson.build | 2 +-
 src/libtracker-fts/meson.build  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libtracker-data/meson.build b/src/libtracker-data/meson.build
index d5c384b2b..ae4487496 100644
--- a/src/libtracker-data/meson.build
+++ b/src/libtracker-data/meson.build
@@ -73,7 +73,7 @@ libtracker_data = library('tracker-data',
     # symbols from libtracker-sparql, but does not mean it should
     # be linked to it.
     override_options: ['b_lundef=false'],
-    dependencies: tracker_data_dependencies + [unicode_library],
+    dependencies: tracker_data_dependencies + [unicode_library, tracker_sparql_intermediate_dep],
     include_directories: [commoninc, configinc, srcinc],
 )
 
diff --git a/src/libtracker-fts/meson.build b/src/libtracker-fts/meson.build
index 5ab4409f8..1392699f8 100644
--- a/src/libtracker-fts/meson.build
+++ b/src/libtracker-fts/meson.build
@@ -11,7 +11,7 @@ libtracker_fts = static_library('tracker-fts',
     'tracker-fts-config.c',
     'tracker-fts-tokenizer.c',
     libtracker_fts_fts5,
-    dependencies: [tracker_common_dep],
+    dependencies: [tracker_common_dep, tracker_sparql_intermediate_dep],
     c_args: tracker_c_args
 )
 
-- 
2.20.1

